### PR TITLE
Improve Kruise E2E image registry and fix image pulling

### PR DESCRIPTION
**Pull request by FillZpp**
Signed-off-by: FillZpp <FillZpp.pub@gmail.com>

### Ⅰ. Describe what this PR does
1. Improve Kruise E2E cases
2. Make E2E image registry to be configurable
3. Fix image pulling for specific images like `docker.io/library/nginx:1.14-alpine`

### CODE REVIEW

File: test/e2e/apps/broadcastjob.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ var _ = SIGDescribe("BroadcastJob", func() {
 
 	framework.KruiseDescribe("BroadcastJob dispatching", func() {
 
-		ginkgo.It("succeeds for parallelism < number of node", func() {
+		framework.ConformanceIt("succeeds for parallelism < number of node", func() {
```
Comment (furykerry) [inline test/e2e/apps/broadcastjob.go:85]: why change the test to conformance test ?

**Review by furykerry — COMMENTED** [review]
/lgtm

File: test/e2e/apps/broadcastjob.go (hunk 1 above)
Comment (yjjy0921) [reply test/e2e/apps/broadcastjob.go:85]: I think the e2e framework is based on k8s e2e framework. But we should not use ConformanceIt here.Need fix it

**Pull request merged**